src/occCAD/G4OCCExplorer.cc

```diff
@@ -0,0 +1,190 @@
+//
+// Created by Boogert, Stewart (-,DL,-) on 27/07/2023.
+//
+
+#include "occCAD/G4OCCExplorer.hh"
+
+#include "XCAFDoc.hxx"
+#include "TDF_Tool.hxx"
+#include "Standard_GUID.hxx"
+#include "TDF_ChildIterator.hxx"
+#include "TopExp_Explorer.hxx"
+#include "TopoDS.hxx"
+#include "BRep_Tool.hxx"
+#include "Geom_Plane.hxx"
+#include "Geom_CylindricalSurface.hxx"
+
+// G4OCCExplorer::G4OCCExplorer() {}
+
+// G4OCCExplorer::~G4OCCExplorer() {}
+
+void G4OCCExplorer::Dump(Handle(XCAFDoc_ShapeTool) shapeTool) {
+    /* Free shapes */
+
+    /* Dump shape tool */
+    shapeTool->Dump(std::cout);
+}
+
+opencascade::handle<TDataStd_Name> G4OCCExplorer::GetNameFromLabel(TDF_Label &label) {
+    auto nameID = TDataStd_Name::GetID();
+    Handle(TDataStd_Name) name;
+    label.FindAttribute(nameID, name);
+    return name;
+}
+
+std::string G4OCCExplorer::GetEntryFromLabel(TDF_Label &label)
+{
+    TCollection_AsciiString anEntry;
+    TDF_Tool::Entry(label,anEntry);
+    return anEntry.ToCString();
+}
+
+opencascade::handle<XCAFDoc_Location> G4OCCExplorer::GetLocationFromLabel(TDF_Label &label) {
+    auto locationID = XCAFDoc_Location::GetID();
+    Handle(XCAFDoc_Location) location;
+    auto b = label.FindAttribute(locationID, location);
+    if(b)
+        return location;
+    else
+        return nullptr;
+}
+
+TDF_Label G4OCCExplorer::FindLabelByEntryString(TDF_Label topLabel, std::string entry) {
+    for (TDF_ChildIterator itall (topLabel,Standard_True); itall.More(); itall.Next()) {
+        auto aChild = itall.Value();
+        if ( GetEntryFromLabel(aChild) == entry) {
+            return aChild;
+        }
+    }
+    return TDF_Label();
+}
+
+TDF_LabelSequence G4OCCExplorer::FindLabelByNameString(TDF_Label topLabel, std::string name) {
+
+    TDF_LabelSequence labelSeq;
+    for (TDF_ChildIterator itall (topLabel,Standard_True); itall.More(); itall.Next()) {
+        auto aChild = itall.Value();
+        if (GetNameFromLabel(aChild)->Get().IsEqual(name.c_str())) {
+            labelSeq.Append(aChild);
+        }
+    }
+    return labelSeq;
+}
+
+void G4OCCExplorer::DumpShape(TopoDS_Shape &shape) {
+    std::cout << "DumpShape" << std::endl;
+
+    auto exp = TopExp_Explorer(shape, TopAbs_FACE, TopAbs_VERTEX);
+
+    while(exp.More()) {
+        auto current = exp.Current();
+
+        //current.DumpJson(std::cout);
```

Comment:
I'd remove commented out code - if you want to leave a note for yourself then write up some comments.